tobac/segmentation/watershed_segmentation.py

```diff
@@ -1267,18 +1292,18 @@ def segmentation(
             segment_number_below_threshold=segment_number_below_threshold,
             statistic=statistic,
         )
-        segmentation_out_list.append(segmentation_out_i)
+        segmentation_out_data.loc[{time_var_name: time_i}] = segmentation_out_i
         features_out_list.append(features_out_i)
         logging.debug(
-            "Finished segmentation for " + time_i.strftime("%Y-%m-%d_%H:%M:%S")
+            "Finished segmentation for "
+            + pd.to_datetime(time_i.values).strftime("%Y-%m-%d %H:%M:%S")
         )
 
     # Merge output from individual timesteps:
-    segmentation_out = segmentation_out_list.merge_cube()
     features_out = pd.concat(features_out_list)
-
+    # segmentation_out = segmentation_out_data.to_dataset()
```

Comment:
Best to remove this line, as having the segmentation mask returned as a dataset would cause issues with the conversion decorators